### PR TITLE
implement pdf preview and other bugs

### LINKED ISSUE #46: Implement PDF preview and fix bugs
Implement PDF preview and fix other bugs.

### PR DESCRIPTION
**Pull request by benedikt-schwering**
close #46 

- pdf preview
- file sorting
- diff selection
- repo card width

### CODE REVIEW

File: ui/src/routes/[repository]/file/[fileVersion]/history/+page.svelte [hunk 1]
```diff
@@ -45,7 +45,7 @@
                 <span class="timestamp">
                     {toTimestamp(fileVersion.id)}
 
-                    <input type=checkbox disabled={fileVersion.deleted || ['png', 'jpg', 'gif'].includes(data.fileVersion.fileName.split('.')[1])} bind:group={$diffIds} value={fileVersion.id}>
+                    <input type=checkbox disabled={fileVersion.deleted || ['png', 'jpg', 'gif', 'pdf'].includes(data.fileVersion.fileName.split('.')[1])} bind:group={$diffIds} value={fileVersion.id}>
```
Comment (boba2fett) [inline ui/src/routes/[repository]/file/[fileVersion]/history/+page.svelte:48]: We should use .at(-1) instead, so files like file.name.png could work too.
also in the other places...

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at

File: ui/package.json [hunk 2]
```diff
@@ -26,6 +26,7 @@
 	},
 	"type": "module",
 	"dependencies": {
-		"monaco-editor": "^0.34.1"
+		"monaco-editor": "^0.34.1",
+		"svelte-pdf": "^1.0.16"
```
Comment (boba2fett) [inline ui/package.json:30]: We don't need this, because we are using the iframe?

**Review by boba2fett — COMMENTED** [review]
Besides the comments, it looks wonderful!

File: ui/src/routes/[repository]/file/[fileVersion]/history/+page.svelte (hunk 1 above)
Comment (DlieBG) [reply ui/src/routes/[repository]/file/[fileVersion]/history/+page.svelte]: solved here: https://github.com/DlieBG/gum/pull/47/commits/238e1b4ca1e10d39fa94dbf222f077ca374886c6

File: ui/package.json (hunk 2 above)
Comment (DlieBG) [reply ui/package.json]: solved here: https://github.com/DlieBG/gum/pull/47/commits/7628aa5c2e3dd08dff9e347f17a4f81d793ce193

**Review by boba2fett — APPROVED** [review]
Wonderful!

**Pull request merged**